paper/src/main/java/com/denizenscript/denizen/paper/events/PrePlayerAttackEntityScriptEvent.java

```diff
@@ -0,0 +1,84 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.ItemTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import io.papermc.paper.event.player.PrePlayerAttackEntityEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.inventory.ItemStack;
+
+public class PrePlayerAttackEntityScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player attacks <entity>
+    //
+    // @Location true
+    //
+    // @Plugin Paper
+    //
+    // @Group Paper
+    //
+    // @Cancellable true
+    //
+    // @Switch with:<item> to only process the event if the player attacks the entity with the specified item.
+    //
+    // @Triggers when the player tries to attack an entity. This occurs before any of the damage logic, so cancelling this event will prevent any sort of sounds from being played when attacking.
+    //
+    // @Context
+    // <context.entity> returns the entity that was attacked in this event.
+    // <context.attacked> returns whether this entity would be attacked normally.
+    // Entities like falling sand would return false because their entity type does not allow them to be attacked.
+    // Note: there may be other factors (invulnerability, etc.) that will prevent this entity from being attacked
+    // that this event will not cover.

```

Comment:
These lines (starting with `Entities like...`) should be moved under `@Triggers` and not be with the contexts. Also, the last line is wrapped a bit weirdly as well and you can totally tack that last one onto the line above it. Generally keep them around 180 characters and then you should try and wrap.